Android/Samples/libraries/moduletools/src/main/java/com/microsoft/maps/moduletools/ParsingHelpers.java

```diff
@@ -32,20 +32,22 @@ public static void logAltitudeWarning() {
   public static String getErrorMessageForPolygonRing(@NonNull ArrayList<Geoposition> positions) {
     if (positions.size() < 4) {
       StringBuilder positionsStringBuilder = new StringBuilder();
-      for (int i = 0; i < positions.size() - 1; i++) {
-        Geoposition position = positions.get(i);
-        positionsStringBuilder
-            .append(
-                "["
-                    + position.getLatitude()
-                    + ", "
-                    + position.getLongitude()
-                    + ", "
-                    + position.getAltitude()
-                    + "]")
-            .append(", ");
+      if (positions.size() > 0) {
+        for (int i = 0; i < positions.size() - 1; i++) {
```

Comment:
Can be a for each loop since the index is not used for anything other than getting the item.

Reply:
Currently, this is not a for each loop, so that the formatting (", ") is not added after the last element. The last element is added manually without the comma.